trieste/models/gpflux/sampler.py

```diff
@@ -331,42 +338,79 @@ def _prepare_weight_sampler(self) -> Callable[[int], TensorType]:  # [B] -> [B,
         canonical features for `P` outputs.
         """
 
-        if isinstance(self._layer.inducing_variable, InducingPoints):
+        if isinstance(self._layer.inducing_variable, SharedIndependentInducingVariables):
             inducing_points = self._layer.inducing_variable.Z  # [M, D]
+            separate_case = False
+        elif isinstance(self._layer.inducing_variable, SeparateIndependentInducingVariables):
+            #inducing_points = tf.self._layer.inducing_variable.Z  # [P, M, D]
+            inducing_points = tf.stack([ind_var.Z for ind_var in self._layer.inducing_variable_list], axis = 0)  # [P, M, D]
+            separate_case = True
         else:
             inducing_points = self._layer.inducing_variable.inducing_variable.Z  # [M, D]
+            separate_case = False
 
         q_mu = self._layer.q_mu  # [M, P]
         q_sqrt = self._layer.q_sqrt  # [P, M, M]
+        
+        #NOTE -- I don't understand why in the original code this approach was used and not the gpflow.covariances.Kuu dispatcher
```

Comment:
leave this as a comment for @sebastianober  here in github rather than as a comment in the code :)